src/Hamiltonians/ExtendedHubbardReal1D.jl

```diff
@@ -13,20 +13,26 @@ Implements the extended Hubbard model on a one-dimensional chain in real space.
 * `u`: on-site interaction parameter
 * `v`: the next-neighbor interaction
 * `t`: the hopping strength
+* `boundary_condition` : applied boundray condition 
+
+# Boundary conditions
+
+* For π-twisted -> :twisted
+* For hard wall -> :hard_wall
 
 """
-struct ExtendedHubbardReal1D{TT,A<:SingleComponentFockAddress,U,V,T} <: AbstractHamiltonian{TT}
+struct ExtendedHubbardReal1D{TT,A<:SingleComponentFockAddress,U,V,T,BOUNDARY_CONDITION} <: AbstractHamiltonian{TT}
     add::A
 end
 
 # addr for compatibility.
-function ExtendedHubbardReal1D(addr; u=1.0, v=1.0, t=1.0)
+function ExtendedHubbardReal1D(addr; u=1.0, v=1.0, t=1.0, boundary_condition = :periodic)
```

Comment:
Add a check here that makes sure the values passed as `boundary_condition` makes sense. Right now, you can pass it anything (e.g. a type) and it will behave as periodic boundaries.

```julia
julia> H1 = ExtendedHubbardReal1D(BoseFS(1,1,1,1))
ExtendedHubbardReal1D(BoseFS{4,4}(1, 1, 1, 1); u=1.0, v=1.0, t=1.0, boundary_condition=:periodic)

julia> H2 = ExtendedHubbardReal1D(BoseFS(1,1,1,1); boundary_condition=:hrad_wall)
ExtendedHubbardReal1D(BoseFS{4,4}(1, 1, 1, 1); u=1.0, v=1.0, t=1.0, boundary_condition=:hrad_wall)

julia> sparse(H1) == sparse(H2)
true
```